sdk/identity/identity/src/credentials/interactiveBrowserCredential.ts

```diff
@@ -104,7 +109,37 @@ export class InteractiveBrowserCredential implements TokenCredential {
   ): Promise<AccessToken | null> {
     const scopeArray = typeof scopes === "object" ? scopes : [scopes];
 
-    return this.acquireTokenFromBrowser(scopeArray);
+    if (this.authenticationRecord && this.persistenceEnabled) {
+      return this.acquireTokenFromCache().catch((e) => {
```

Comment:
not sure why this method isn't async so you can try/catch here instead of using `.catch`